server/src/main/java/com/objectcomputing/checkins/services/guild/GuildController.java

```diff
@@ -41,59 +56,95 @@ public GuildController(GuildServices guildService) {
+                    //Using code block rather than lambda so we can log what thread we're in
+                    return (HttpResponse<Guild>) HttpResponse
+                            .created(createdGuild)
+                            .headers(headers -> headers.location(
+                                    URI.create(String.format("%s/%s", request.getUri(), createdGuild.getId()))));
+                }).subscribeOn(Schedulers.from(ioExecutorService));
 
-    @Post(value = "/")
-    public HttpResponse<Guild> createAGuild(@Body @Valid GuildCreateDTO guild, HttpRequest<GuildCreateDTO> request) {
-        Guild newGuild = guildService.save(new Guild(guild.getName(), guild.getDescription()));
-        return HttpResponse
-                .created(newGuild)
-                .headers(headers -> headers.location(URI.create(String.format("%s/%s", request.getUri(), newGuild.getId()))));
     }
 
-    /**
-     * Load the current guilds into checkinsdb.
-     *
-     * @param guildsList, array of {@link GuildCreateDTO guild create dto} to load {@link Guild guild(s)}
-     */
-
-    @Post("/guilds")
-    public HttpResponse<?> loadGuilds(@Body @NotNull @Valid List<GuildCreateDTO> guildsList, HttpRequest<List<GuildCreateDTO>> request) {
-        List<String> errors = new ArrayList<>();
-        List<Guild> guildsCreated = new ArrayList<>();
-        for (GuildCreateDTO guildDTO : guildsList) {
-            Guild guild = new Guild(guildDTO.getName(), guildDTO.getDescription());
-            try {
-                guildService.save(guild);
-                guildsCreated.add(guild);
-            } catch (GuildBadArgException e) {
-                errors.add(String.format("Guild %s was not added because: %s", guild.getName(), e.getMessage()));
-            }
-        }
-        if (errors.isEmpty()) {
-            return HttpResponse.created(guildsCreated).headers(headers ->
-                    headers.location(request.getUri()));
-        } else {
-            return HttpResponse.badRequest(errors).headers(headers ->
-                    headers.location(request.getUri()));
-        }
-    }
+     /**
+      * Load the current guilds into checkinsdb.
+      *
+      * @param guildsList, array of {@link GuildCreateDTO guild create dto} to load {@link Guild guild(s)}
+      */
+     @Post("/guilds")
+     public Single<HttpResponse<?>> loadGuilds(@Body @NotNull @Valid List<GuildCreateDTO> guildsList, HttpRequest<List<GuildCreateDTO>> request) {
+
+         return Single.fromCallable(() -> {
+             List<String> errors = new ArrayList<>();
+             List<Guild> guildCreated = new ArrayList<>();
+             for (GuildCreateDTO guildDTO : guildsList) {
+                 Guild guild = new Guild(guildDTO.getName(),
+                         guildDTO.getDescription());
+                 try {
+                     guildService.save(guild);
+                     guildCreated.add(guild);
+                 } catch (CompositeException e) {
```

Comment:
what's the advantage of throwing composite exception here ? when we call save only GuildBadArg exception is thrown in service class. I am new to RxJava so trying to understand why it is done this way ? @meyerinj 
@Maroju100 also there are no test cases to test this exception ?  how are we going to test this ? because right now it just throws BadArg exception from service layer as per the test cases we have.

Reply:
There are scenarios where a reactivex process, because it can encompass multiple threads, may need to throw multiple exceptions. This gets bundled up in a composite exception. This try/catch occurs within a single reactivex process, so it shouldn't ever have the opportunity catch a composite exception. Rather, it's something that could come up if the entire `Single.fromCallable` was wrapped in a try/catch. This is why many of the other controllers have an error handler for composite exception

Reply:
Based on chats with John and Michael we have decided to remove the bulk load end points and the associated tests, created an issue for it.

Reply:
Good this answer during sprint planning, will be removing this on team.